electron_app/src/electron-main.js

```diff
@@ -267,18 +276,36 @@ ipcMain.on('seshat', async function(ev, payload) {
 
     switch (payload.name) {
         case 'supportsEventIndexing':
-            if (Seshat === null) ret = false;
-            else ret = true;
+            ret = seshatSupported;
             break;
 
         case 'initEventIndex':
             if (eventIndex === null) {
                 try {
                     await afs.mkdir(eventStorePath, {recursive: true});
-                    eventIndex = new Seshat(eventStorePath, {passphrase: "DEFAULT_PASSPHRASE"});
+                    eventIndex = new Seshat(eventStorePath, {passphrase: seshatPassphrase});
                 } catch (e) {
-                    sendError(payload.id, e);
-                    return;
+                    if (e instanceof ReindexError) {
+                        // If this is a reindex error, the index schema
+                        // changed. Try to open the database in recovery mode,
+                        // reindex the database and finally try to open the
+                        // database again.
+                        try {
+                            recoveryIndex = new SeshatRecovery(eventStorePath,
+                                                               {passphrase: seshatPassphrase});
+                            await recoveryIndex.reindex();
+                            // Delete the recovery index here to make sure that
+                            // the locks are freed.
+                            delete recoveryIndex;
```

Comment:
the delete operator in js is for deleting an array element rather than a free() style thing - unsure what the effect of this would be - probably nothing. If you want to explicitly release resources afaik you'll have to do it with an explicit method (eg. `destroy()` or something) rather than rely on a destructor.

Reply:
The `reindex()` already consumes the database object and executes a `drop()` on the rust side of things so I think this is a thinko on my side and we can just remove the comment and the `delete`.